releasenotes/notes/pilot-load-dns-cert-known-location-deprecate-flags.yaml

```diff
@@ -0,0 +1,46 @@
+apiVersion: release-notes/v2
+
+# This YAML file describes the format for specifying a release notes entry for Istio.
+# This should be filled in for all user facing changes.
+
+# kind describes the type of change that this represents.
+# Valid Values are:
+# - bug-fix -- Used to specify that this change represents a bug fix.
+# - security-fix -- Used to specify that this change represents a vulnerability fix.
+# - feature -- Used to specify a new feature that has been added.
+# - test -- Used to describe additional testing added. This file is optional for
+#   tests, but included for completeness.
+kind: feature
+# area describes the area that this change affects.
+# Valid values are:
+# - traffic-management
+# - security
+# - telemetry
+# - installation
+# - istioctl
+# - documentation
+area: security
+
+# issue is a list of GitHub issues resolved in this note.
+# If issue is not in the current repo, specify its full URL instead.
+issue:
+  - 36916
+
+
+# releaseNotes is a markdown listing of any user facing changes. This will appear in the
+# release notes.
+releaseNotes:
+- |
+  **Changed** Pilot will now load its DNS serving certificate from well known locations:
+```
+/etc/pilot/tls/tls.crt
+/etc/pilot/tls/tls.key
+/etc/pilot/ca/root-cert.pem
```

Comment:
What's the motivation for doing this?

Reply:
I think it might be helpful to step back in history here :)
Initially we wanted to expose the tls pilot dns cert flags on the helm side, this was to allow helm users to install istio-csr with helm istio, which is currently done with overlays in istio operator:
[example](https://github.com/cert-manager/istio-csr/blob/main/hack/istio-config-1.13.4.yaml)

This was not only crucial for helm istio installation users, but for Maistra Openshift service mesh, where we're trying to have better integration, but under the hood their operator mirrors the istio helm charts.

After doing the initial work of exposing the tls args, Costin thought we could simplify the configuration of istio-csr and make it 0 conf, like it is for gateway ssl certs etc.. how this would work is to load the volumes that istio-csr makes for pilot tls cert but with optional:true , and add some detection logic , just like you can see already implemented in PR, this would mean that if using istio-csr you just need to install istio-csr, say Citadel disabled, point CA_ADDRESS to istio-csr and everything would work fine.
 
https://github.com/istio/istio/pull/38845#issuecomment-1141234213

As to the reason why we have to load the external tls cert in the first place that is because our external CA signers need to talk to a pilot that is signed by same CA, so there is a chicken and egg for trust, so we need to load the pilot cert at installation time for istiod.

Hope that makes sense!